INSS.EIIR.DataSync.Infrastructure/Source/SQL/INSSightSQLSource.cs

```diff
@@ -0,0 +1,37 @@
+﻿using INSS.EIIR.DataSync.Application.UseCase.SyncData.Infrastructure;
+using INSS.EIIR.DataSync.Application.UseCase.SyncData.Model;
+using INSS.EIIR.DataSync.Infrastructure.Source.SQL.Context;
+using INSS.EIIR.Models.CaseModels;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace INSS.EIIR.DataSync.Infrastructure.Source.SQL
+{
+    public class INSSightSQLSource : IDataSourceAsync<InsolventIndividualRegisterModel>
+    {
+        private readonly ExternalInssContext _externalInssContext;
+        private readonly SQLSourceOptions _options;
+
+        public INSSightSQLSource(SQLSourceOptions options) 
+        {
+            var dbContextOptsBldr = new DbContextOptionsBuilder<ExternalInssContext>();
+            dbContextOptsBldr.UseSqlServer(options.ConnectionString, sqlServerOptions => sqlServerOptions.CommandTimeout(60));
+
+            _externalInssContext = new ExternalInssContext(dbContextOptsBldr.Options);
+            this._options = options;
+        }
+
+        public async IAsyncEnumerable<InsolventIndividualRegisterModel> GetInsolventIndividualRegistrationsAsync()
+        {
+            await foreach (var x in _externalInssContext.VwEiirs.Where(p => p.InsolvencyType == "Bankruptcy" && p.HasRestrictions == 1).AsNoTracking().AsAsyncEnumerable())
```

Comment:
Use Constants for the string "Bankruptcy"

Reply:
Ultimately won't be required as this code is just temporary to reduce the dataset retrieved from INSSight in short term added a commentto explain why